smarty/forcefield.py

```diff
@@ -671,11 +717,66 @@ def createSystem(self, topology, molecules, nonbondedMethod=NoCutoff, nonbondedC
 
         return system
 
+    def labelMolecules(self, oemols, verbose = False):
+        """Return labels for a list of OEMols corresponding to parameters from this force field. For each oemol, a dictionary of force types is returned, and for each force type, each force term is provided with the atoms involved, the parameter id assigned, and the corresponding SMIRKS.
+
+        Parameters
+        ----------
+        oemols : list of OEMols
+            The OpenEye OEChem OEMol objects as a list; these will be labeled. Should include all atoms with the correct ordering atom atom numbers will be returned along with corresponding labeling.
+        verbose : bool
+            If True, verbose output will be printed
+
+        Returns
+        -------
+        molecule_labels : list
+            list of labels for molecules. Each entry in the list corresponds to
+            one molecule from the provided list of oemols and is a dictionary
+            keyed by force type, i.e. molecule_labels[0]['HarmonicBondForce']
+            gives details for the harmonic bond parameters for the first
+            molecule. Each element is a list of the form [ ( [ atom1, ...,
+            atomN], parameter_id, SMIRKS), ... ]
+
+        """
+
+        # XML modified? If so, re-parse by generators
+        if self._XMLModified:
+            if verbose: print("Re-parsing XML because it was modified.")
+            self.parseXMLTrees()
+
+        # Storage for labels
+        molecule_labels = []
+
+        # Loop over molecules and label
+        for idx,mol in enumerate(oemols):
+            molecule_labels.append({})
+            for force in self._forces:
+                # Initialize dictionary storage for this force type
+                if isinstance(force, HarmonicBondGenerator):
```

Comment:
For example, you could just use `force.__class__.__name__` if you are OK with `HarmonicBondGenerator`, or you can add a class field to each force generator like `label = 'HarmonicBondForce'`.